### ISSUE TITLE
modelviewer ar-placement= wall is flipping the glb model

### ISSUE THREAD

**Issue by anil-arnxt** [google/model-viewer#4771]
### Description

 I am using modelviewer to show glb model of an AC on the wall but its flipping it 360 degree when i place it on the wall.

### Version
   
![Screenshot_2024-05-06-12-29-24-231_com android chrome](https://github.com/google/model-viewer/assets/121437693/703046a6-7dd8-4647-99f2-6b8236605320)

### Browser Affected

- [ ] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by vincentfretin** [comment]
This is discussed in #3989

**Comment by anil-arnxt** [comment]
Thanks for your reply. But I am working on Android not ios.

**Comment by anil-arnxt** [comment]
https://github.com/google/model-viewer/assets/121437693/34fe3201-1d12-4312-8204-7d7369ffb332

**Comment by vincentfretin** [comment]
From the video, it seems to be an issue with SceneViewer on Android. In this case please check only the correct checkboxes on the issue. Not sure if we can do anything about it.
You can set `ar-modes="webxr quick-look"` to force using webxr and not scene-viewer on Android. I didn't have an issue with webxr mode on Android.

**Comment by anil-arnxt** [comment]
<model-viewer
                        id="change-speed-demo"
                        camera-controls
                        touch-action="pan-y"
                        src={glbFile}
                        ar
                      ar-placement={placement}
                  
                        alt="A 3D model of a helmet"
                        ref={modelViewerRef}
                        animation-name="Dance"
                        ar-modes="webxr scene-viewer quick-look"
                        shadow-intensity="1">
                        <button
                          slot="ar-button"
                          id="ar-button"
                          onClick={handlearclick}>
                          View in your space
                        </button>
                          {showOverlay && (
                        <div className="overlay">
                          <p>Move your device to place the object in AR</p>
                          <button onClick={handlePlaceObject}>Start AR</button>
                        </div>
                      )}
                      <div id="ar-prompt">
                        <img src="https://modelviewer.dev/shared-assets/icons/hand.png" />
                        <div id="wall_floor">
                          Move Camera on the <span>{placement}</span>
                        </div>
                      </div>

                      <div id="ar-failure">AR is not tracking!</div>
                        {hasAnimation && (
                          <div id="controls">
                            <button onClick={handleToggleAnimation}>
                              {isPlaying ? <FaPause /> : <FaPlay />}
                            </button>
                          </div>
                        )}
                      </model-viewer>  

  this is my code for ios its not flipping but when opening on android its flipping the model.

**Comment by elalish** [comment]
No, that's WebXR mode alright. Weird! Does that repro reliably even on other walls? The only thing I can imagine is that the returned hit matrix has an inverted normal. @bialpio Have you ever seen anything like this, or have a thought as to where in the code that could happen? The relevant piece of MV code is:

```
    if (this.placeOnWall === true) {
      // Orient the model to the wall's normal vector.
      this.goalYaw = Math.atan2(hitMatrix.elements[4], hitMatrix.elements[6]);
    }
```
which I don't see any problems with. `hitMatrix` is returned right from the WebXR API.

**Comment by elalish** [comment]
@anil-arnxt Can you fill in the device and OS/browser version info so we can try to repro?

**Comment by anil-arnxt** [comment]
@elalish I am using Android version 12 and chrome browser.

**Comment by elalish** [comment]
@anil-arnxt Can we have more details on your device type and browser version? (the xx.x.xxxx.xxx are intended to be filled in by you as part of the issue template).

**Comment by anil-arnxt** [comment]
@elalish please see my device that is mobile details
![Screenshot_2024-05-16-10-50-01-027_com android settings](https://github.com/google/model-viewer/assets/121437693/3e90b27f-945f-4c15-ad93-23c5b1c8f7d2)
![Screenshot_2024-05-16-10-49-54-807_com android settings](https://github.com/google/model-viewer/assets/121437693/bb960078-c0b6-4870-98e4-79de84a25b6b)
![Screenshot_2024-05-16-10-50-23-641_com android chrome](https://github.com/google/model-viewer/assets/121437693/cd7493cc-f2a7-4ecf-a0e5-b14c64d80931)

**Comment by luciousBrown** [comment]
Hi, I am also having this issue and noticed it started happening about a week ago for me with projects that I had made no changes to and that never had this problem before. As soon as i place an object on the wall such as a painting, it immediately flips 180 degrees and is now hard to rotate to correct face. I am using webxr and not scene viewer and as mentioned, this issue did not exist before and is also happening with the wall placement example provided on your site https://modelviewer.dev/examples/augmentedreality/#wall. Would like to know how this can be fixed as it is an issue with commercial clients intending to use MV for wall placements. Thanks.

**Comment by elalish** [comment]
You're quite correct - @devbridie has verified this is a regression in the latest version of ARCore. Working on a fix, but not sure exactly when it'll roll out yet.

**Comment by luciousBrown** [comment]
Hi @elalish thanks for looking into this. Have had to put a pause on some client projects as even though can use Sceneviewer, it is not as good as modelviewer in my opinion in regards to lighting and customization etc. iOS quicklook users are having no problems though.

**Comment by devbridie** [comment]
This issue will be resolved in ARCore 1.44 which is expected to go out in a few weeks.

**Comment by luciousBrown** [comment]
Hi @devbridie thanks for the update. Looking forward to the fix.

**Comment by devbridie** [comment]
Hey, would love to hear if the recent updates have resolved your issue. Please verify that Google Play Services for AR is version 1.44 on the target devices and let me know if it's fixed!

**Comment by milesgreen** [comment]
Just updated to 1.44 and can confirm that, for me at least, I'm seeing an improvement. Vertically anchored models are no longer flipping for me.
